### ISSUE TITLE
some issue with iio_create_default_context

### ISSUE THREAD

**Issue by rgetz** [analogdevicesinc/libiio#301]
[iio_create_default_context](https://analogdevicesinc.github.io/libiio/group__Context.html#ga5f18bbe6c9e1993f26480f73f92e1ad5) says it is supposed to 
- create a network context if the IIOD_REMOTE environment variable
-  the server will be discovered using ZeroConf (if set to an empty string)

This works:
```
rgetz@brain:~/github/libiio/examples$ IIOD_REMOTE="pluto.local" ./ad9361-iiostream -h
* Acquiring IIO context
* Acquiring AD9361 streaming devices
* Configuring AD9361 for streaming
* Acquiring AD9361 phy channel 0
* Acquiring AD9361 RX lo channel
* Acquiring AD9361 phy channel 0
* Acquiring AD9361 TX lo channel
* Initializing AD9361 IIO streaming channels
* Enabling IIO streaming channels
* Creating non-cyclic IIO buffers with 1 MiS
* Starting IO streaming (press CTRL+C to cancel)
	RX     1.05 MSmp, TX     1.05 MSmp
	RX     2.10 MSmp, TX     2.10 MSmp
	RX     3.15 MSmp, TX     3.15 MSmp
	RX     4.19 MSmp, TX     4.19 MSmp
^CWaiting for process to finish...
	RX     5.24 MSmp, TX     5.24 MSmp
* Destroying buffers
* Disabling streaming channels
* Destroying context
```

This fails:
```
rgetz@brain:~/github/libiio/examples$ IIOD_REMOTE="" ./ad9361-iiostream -h
* Acquiring IIO context
* Acquiring AD9361 streaming devices
* Configuring AD9361 for streaming
* Acquiring AD9361 phy channel 0
* Acquiring AD9361 RX lo channel
* Acquiring AD9361 phy channel 0
* Acquiring AD9361 TX lo channel
* Initializing AD9361 IIO streaming channels
* Enabling IIO streaming channels
* Creating non-cyclic IIO buffers with 1 MiS
Could not create TX buffer: Device or resource busy
* Destroying buffers
* Disabling streaming channels
* Destroying context
```

according to the doc, they should both be the same. (tested on Pluto).

**Comment by rgetz** [comment]
I think this might be related to #307 as well...

Does anyone want to track this down? Or those that found it, fix it?

-Robin

**Comment by rgetz** [comment]
The fix described in #307 fixes this as well...

IIOD_REMOTE="" ./tests/iio_attr -d
IIO context has 5 devices:
	iio:device0: adm1177, found 0 device attributes
	iio:device1: ad9361-phy, found 18 device attributes
	iio:device2: xadc, found 1 device attributes
	iio:device3: cf-ad9361-dds-core-lpc, found 0 device attributes
	iio:device4: cf-ad9361-lpc, found 0 device attributes


so closing this
